make receiving events nonblocking even when queue is full

adamdubiel opened the pull request:
Also added two metrics:

* discarded (dropped) events when queue is full
* event queue utilization on 0-100 scale

janisz commented on web/event_handler.go:
```diff
@@ -20,20 +20,22 @@ type event struct {
+	serviceRegistry    service.ServiceRegistry
+	marathon           marathon.Marathoner
+	eventQueue         <-chan event
+	eventQueueCapacity int
```
Do we really need this value? Can't just use [cap](https://golang.org/pkg/builtin/#cap)

adamdubiel replied:
oh great, this is what i was looking for and did not find, will change

adamdubiel commented:
Are build always this flaky?

janisz commented:
Tests get flaky when they are run in parallel (multiple packages at once) but in Travis we have run them sequentially. 
Unfortunately your code doesn't pass linter https://travis-ci.org/allegro/marathon-consul/builds/186278642#L217

adamdubiel commented:
Oh, okay, linter this time. But if you see build history, each time different tests were failing (build continues after 1st fail).

adamdubiel commented:
Found two places that made tests at my localhost (Mac) fail:

* bumped Consul version to 0.7.2, which fixed issue with Consul segfaults at random moments
* fixed race condition detected by `-race` in Marathon stub

I also added 2 new targets to Makefile:

* `format` that runs `go fmt` on `marathon-consul` sources
* `check` that runs `gometalinter` in same config as on travis

janisz commented on web/event_handler.go:
```diff
@@ -55,7 +55,11 @@ func (fh *eventHandler) start() chan<- stopEvent {
 				metrics.Mark(fmt.Sprintf("events.handler.%d", fh.id))
-				metrics.UpdateGauge("events.queue.len", int64(len(fh.eventQueue)))
+
+				ql := int64(len(fh.eventQueue))
```
Use meaningful name for variable `queueLength`

adamdubiel replied:
Yeah, looks like the whole go world likes to use those silly one or two letter names, just tried to follow ;) Will change

janisz commented on web/event_handler_test.go:
```diff
@@ -20,7 +20,7 @@ func TestWebHandler_DropUnknownEventType(t *testing.T) {
 
 	// given
-	queue := make(chan event)
+	queue := make(chan event, 10)
```
Why you use buffered channel?

adamdubiel replied:
might not make sense anymore, but it helped to pass any tests at the beginning. will change

adamdubiel commented:
i'm going to close this pr and try to work on tests. they fail randomly both on localhost and travis when using nonblocking channel.
